contracts/src/node-interface/NodeInterface.sol

```diff
@@ -123,4 +123,9 @@ interface NodeInterface {
             uint256 amount,
             bytes memory calldataForL1
         );
+
+    // @notice Returns the first block produced using the Nitro codebase
+    // @dev returns 0 for chains like Nova that don't contain classic blocks
+    // @return number the block number
+    function nitroGenesisBlock() external pure returns (uint256 number);
```

Comment:
What's the thought on having this be pure vs view?

Reply:
The reason is that `nitroGenesisBlock` doesn't depend on state changes, and there's no guarantee that a `pure` function on one chain will return the same result on another